src/webAppCreator.ts

```diff
@@ -0,0 +1,484 @@
+        super(output);
+        this.steps.push(new SubscriptionStep(this, azureAccount));
+        this.steps.push(new ResourceGroupStep(this, azureAccount));
+        this.steps.push(new AppServicePlanStep(this, azureAccount));
+        this.steps.push(new WebsiteStep(this, azureAccount));
+    }
+
+    protected beforeExecute(step: WizardStep, stepIndex: number) {
+        if (stepIndex == 0) {
+            this.writeline('Start creating new Web App...');
+        }
+    }
+
+    protected onExecuteError(step: WizardStep, stepIndex: number, error: Error) {
+        this.writeline(`Failed to create new Web App - ${error.message}`);
+    }
+
+    protected onComplete() {
+        this.writeline('Done!');
+    }
+}
+
+class SubscriptionBasedWizardStep extends WizardStep {
+    protected constructor(wizard: WizardBase, stepTitle: string, readonly azureAccount: AzureAccountWrapper) {
+        super(wizard, stepTitle);
+    }
+
+    protected getSelectedSubscription(): SubscriptionModels.Subscription {
+        const subscriptionStep = <SubscriptionStep>this.wizard.findStep(step => step instanceof SubscriptionStep, 'The Wizard must have a SubscriptionStep.');
+
+        if (!subscriptionStep.subscription) {
+            throw new Error('A subscription must be selected first.');
+        }
+
+        return subscriptionStep.subscription;
+    }
+
+    protected getSelectedResourceGroup(): ResourceModels.ResourceGroup {
+        const resourceGroupStep = <ResourceGroupStep>this.wizard.findStep(step => step instanceof ResourceGroupStep, 'The Wizard must have a ResourceGroupStep.');
+
+        if (!resourceGroupStep.resourceGroup) {
+            throw new Error('A resource group must be selected first.');
+        }
+
+        return resourceGroupStep.resourceGroup;
+    }
+}
+
+class SubscriptionStep extends WizardStep {
+    private _subscription: SubscriptionModels.Subscription;
+
+    constructor(wizard: WizardBase, readonly azureAccount: AzureAccountWrapper) {
+        super(wizard, 'Select subscription');
+    }
+
+    async prompt(): Promise<void> {
+        const inFilterSubscriptions = await this.azureAccount.getFilteredSubscriptions();
+        const otherSubscriptions = await this.azureAccount.getAllSubscriptions();
+        const quickPickItems: vscode.QuickPickItem[] = [];
+        const quickPickOptions = { placeHolder: `Select the subscription where the new Web App will be created in. (${this.stepProgressText})` };
+
+        inFilterSubscriptions.forEach(s => {
+            const index = otherSubscriptions.findIndex(other => other.subscriptionId === s.subscriptionId);
+            if (index >= 0) {   // Remove duplicated items from "all subscriptions".
+                otherSubscriptions.splice(index, 1);
+            }
+
+            const item = {
+                label: `📌 ${s.displayName}`,
+                description: '',
+                detail: s.subscriptionId
```

Comment:
- Subscription ID is a GUID which is hardly "description" of the subscription.
- I tried both, having the subscription ID on a separate line looks better, especially when the subscription display name is long.

Reply:
"Looks better" is pretty arbitrary. I agree it looks a little cleaner on a separate line, but I don't like the fact that I can only see half my subscriptions at once.

If you can show me other examples where a quick pick uses just label and detail I'm fine with this, otherwise I would rather stick to the more common pattern (For example try 'Git merge' from the command palette and you'll see them use label/description)

Reply:
Just noticed the sign-in extension uses description, too. Will change to match in a separate PR.